extensions/amp-subscriptions/0.1/local-subscription-platform-iframe.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {Entitlement} from './entitlement';
+import {LocalSubscriptionBasePlatform}
+  from './local-subscription-platform-base';
+import {Messenger} from '../../amp-access/0.1/iframe-api/messenger';
+import {assertHttpsUrl, parseUrlDeprecated} from '../../../src/url';
+import {devAssert, userAssert} from '../../../src/log';
+import {isArray} from '../../../src/types';
+import {parseJson} from '../../../src/json';
+import {toggle} from '../../../src/style';
+
+
+/**
+ * Implments the iframe local subscriptions platform which provides
+ * authorization and pingback via an iframe
+ *
+ * @implements {./subscription-platform.SubscriptionPlatform}
+ */
+export class LocalSubscriptionIframePlatform
+  extends LocalSubscriptionBasePlatform {
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!JsonObject} platformConfig
+   * @param {!./service-adapter.ServiceAdapter} serviceAdapter
+   */
+  constructor(ampdoc, platformConfig, serviceAdapter) {
+    super(ampdoc, platformConfig, serviceAdapter);
+
+    devAssert(this.serviceConfig_['type'] == 'iframe',
+        'iframe initialized called without iframe config type');
+
+    /** @const @private {string} */
+    this.iframeSrc_ = userAssert(this.serviceConfig_['iframeSrc'],
+        '"iframeSrc" URL must be specified');
+    assertHttpsUrl(this.iframeSrc_, 'iframe Url');
+
+
+    /** @const @private {?Array} */
+    this.iframeVars_ = this.serviceConfig_['iframeVars'] || null;
+    if (this.iframeVars_) {
+      userAssert(isArray(this.iframeVars_),
+          '"iframeVars" must be an array');
+    }
+
+    /** @private @const {string} */
+    this.targetOrigin_ = parseUrlDeprecated(this.iframeSrc_).origin;
+
+    /** @private {?function()} */
+    this.connectedResolver_ = null;
+
+    /** @private {?Promise} */
+    this.connectedPromise_ = null;
+
+    /** @private @const {!Element} */
+    this.iframe_ = ampdoc.win.document.createElement('iframe');
+    toggle(this.iframe_, false);
```

Comment:
This is actually a pretty interesting possibility here. Definitely for a future discussion, but we could allow an iframe here to also act as a UI provider, just like we do with SwG case.

Reply:
added todo